Lib/functools.py

```diff
@@ -992,21 +989,15 @@ def __get__(self, instance, owner=None):
                 f"instance to cache {self.attrname!r} property."
             )
             raise TypeError(msg) from None
-        val = cache.get(self.attrname, _NOT_FOUND)
-        if val is _NOT_FOUND:
-            with self.lock:
-                # check if another thread filled cache while we awaited lock
-                val = cache.get(self.attrname, _NOT_FOUND)
-                if val is _NOT_FOUND:
-                    val = self.func(instance)
-                    try:
-                        cache[self.attrname] = val
-                    except TypeError:
-                        msg = (
-                            f"The '__dict__' attribute on {type(instance).__name__!r} instance "
-                            f"does not support item assignment for caching {self.attrname!r} property."
-                        )
-                        raise TypeError(msg) from None
+        val = self.func(instance)
```

Comment:
if there are competing threads trying to access the property on first access at the same time, then the property function can be called many times/more than once here..

don't we want to protect against that by design ?

Reply:
@gst you may want to read the Discourse thread linked in the pull request description for some background here.

Reply:
or just read the documentation notes added in this very PR :)

Reply:
well I've read that. 

so this is taking the risk to have previous code which was "guaranted" to run/execute only once the property function, now will be able to run it more than once that same function ?

Reply:
correct